src/components/ScriptExplorer/SignatureImporter.jsx

```diff
@@ -293,83 +308,212 @@ class SignatureImporter extends React.Component {
-        return;
-      }
-      if (publicKey) {
-        for (
-          let finalizedSignatureImporterNum = 0;
-          finalizedSignatureImporterNum < finalizedSignatureImporters.length;
-          finalizedSignatureImporterNum += 1
-        ) {
-          const finalizedSignatureImporter =
-            finalizedSignatureImporters[finalizedSignatureImporterNum];
+    if (numSignatureSets > Object.values(signatureImporters).length) {
+      errback(
+        "Too many signatures. Max one set of signatures per required signer."
+      );
+      return;
+    }
+    const publicKeys = [];
+    if (numSignatureSets === 1) {
+      const finalizedSignatureImporters = Object.values(
+        signatureImporters
+      ).filter((signatureImporter) => signatureImporter.finalized);
+      for (
+        let inputIndex = 0;
+        inputIndex < inputsSignatures.length;
+        inputIndex += 1
+      ) {
+        const inputNumber = inputIndex + 1;
+        const inputSignature = inputsSignatures[inputIndex];
+        if (validateHex(inputSignature) !== "") {
+          errback(`Signature for input ${inputNumber} is not valid hex.`);
+          return;
+        }
 
-          if (
-            finalizedSignatureImporter.signature[inputIndex] ===
-              inputSignature ||
-            finalizedSignatureImporter.publicKeys[inputIndex] === publicKey
+        let publicKey;
+        try {
+          publicKey = validateMultisigSignature(
+            network,
+            inputs,
+            outputs,
+            inputIndex,
+            inputSignature
+          );
+        } catch (e) {
+          errback(`Signature for input ${inputNumber} is invalid.`);
+          return;
+        }
+        if (publicKey) {
+          for (
+            let finalizedSignatureImporterNum = 0;
+            finalizedSignatureImporterNum < finalizedSignatureImporters.length;
+            finalizedSignatureImporterNum += 1
           ) {
-            errback(
-              `Signature for input ${inputNumber} is a duplicate of a previously provided signature.`
-            );
+            const finalizedSignatureImporter =
+              finalizedSignatureImporters[finalizedSignatureImporterNum];
+
+            if (
+              finalizedSignatureImporter.signature[inputIndex] ===
+                inputSignature ||
+              finalizedSignatureImporter.publicKeys[inputIndex] === publicKey
+            ) {
+              errback(
+                `Signature for input ${inputNumber} is a duplicate of a previously provided signature.`
+              );
+              return;
+            }
+          }
+          publicKeys.push(publicKey);
+        } else {
+          errback(`Signature for input ${inputNumber} is invalid.`);
+          return;
+        }
+      }
+      setComplete(number, {
+        signature: inputsSignatures,
+        publicKeys,
+        finalized: true,
+      });
+    } else {
```

Comment:
This whole method feels like it would be useful as it's own utility. I think this will be easier to do once we start settling more on the idea of SignatureSets as its own class/object and have better interactions around PSBTs. For a TODO of course, but this feels useful and also like too much logic to be having in a view/component.